docs/modules/ROOT/pages/explanations/decisions/maintenance-alerts.adoc

```diff
@@ -0,0 +1,84 @@
+= Group Maintenance Alerts
+
+== Problem
+
+Unattended maintenance for OpenShift 4 should just be happening.
+OnCall shouldn't be alerted all the time, instead relevant alerts should be aggregated.
+There's multiple potential approaches to do so, see section <<_proposals,proposals>>
+
+=== Goals
+
+* Alert if automated maintenance of any OpenShift 4 cluster is blocked at any point
+* Respect OnCall engineers sanity and alert only once or as few times as possible
+** Aggregate individual cluster alerts into one single alert
+** Only send out alerts if any alert is firing for a certain period of time
+** Suppress cluster alerts during the maintenance window
+* SLA relevant alerts shouldn't be suppressed in any form
+
+=== Non-goals
+
+* General centralized alerts of OpenShift 4 clusters
+
+== Proposals
+
+=== Option 1: Use centralized Mimir / Grafana
+
+The upgrade controller is monitoring the clusters health and can emit metrics on the current state of the maintenance process.
+We can send these few metrics to our centralized Mimir instance and implement alerting there.
+
+Alternatively record rules could be used to create necessary metric time series.
+The Prometheus `ALERTS` metric is also a record rule under the hood.
+It should be possible to remote write this metric to our centralized Mimir instance.
+This would allow us to build alerting dashboards and meta alerts with minimal additional work and transmitted data.
```

Comment:
Does Mimir have inbuilt alert rule evaluation, or would we need to deploy a Thanos Ruler using metrics from Mimir?

Reply:
The production-grade Mimir deployment comes with it's own mimir-ruler and alertmanager instances